### ISSUE TITLE
Email Link is not working?

### ISSUE THREAD

**Issue by EskAere** [koji-1009/flutter_auth_ui#66]
I've setup Email Link in the past with Firebase Auth, and it was working fine.

I've discovered this plugin (thank you for that) and want to try it out! Google and Facebook work fine. However, I can't make the email link work properly. 

I receive the email, but once I click on the email link and get back to the app, nothing happens! 

Here is my code : 
` emailAuthOption: EmailAuthOption(
        enableMailLink: true,
        handleURL: 'https://myapp.page.link',
        androidPackageName: 'com.mycompany.myapp',
        androidMinimumVersion: '1',
      ),`

That's the only error I have : 
`I/flutter ( 6089): flutter_auth_ui: error => PlatformException(0, error result, null, null)`

Let me know if it's a known issue or if I'm missing something

**Comment by koji-1009** [comment]
@EskAere 
I'm sorry, there seems to be a problem.
Does it work with 2.2.0?

**Comment by koji-1009** [comment]
Would you tell us about the following questions?

* Is this issue happening on iOS?
* Do options such as "https://myapp.page.link" contain the values ​​set on the firebase console?

**Comment by EskAere** [comment]
@koji-1009  

>  Does it work with 2.2.0?
If I haven't made a silly mistake, no

>Is this issue happening on iOS?
I don't have an iOS env.


> Do options such as "https://myapp.page.link" contain the values ​​set on the firebase console? 
Yes everything is setup properly. I've manually implemented dynamic links and it is working. But ideally I'd like to avoid doing the custom auth ui

**Comment by koji-1009** [comment]
Oops, the log format is certainly Android logcat.

**Comment by koji-1009** [comment]
I figured out the problem. It seems that there is not enough implementation to receive deep link on Android, so I will support it.

**Comment by koji-1009** [comment]
@EskAere 

Problem solved!
It needs some more time before it is released, as it needs to be reviewed and the README updated.

If you want to check it at hand, please try the following changes.

`pubspec.yaml`

```yaml
  flutter_auth_ui:
    git:
      url: https://github.com/koji-1009/flutter_auth_ui
      path: flutter_auth_ui
      ref: android_emaillink
```

`MainActivity.java`

```java
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.dr1009.app.flutter_auth_ui.FlutterAuthUiPlugin;

import io.flutter.embedding.android.FlutterActivity;

public class MainActivity extends FlutterActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // check intent
        FlutterAuthUiPlugin.catchEmailLink(this, getIntent());
    }

    @Override
    protected void onNewIntent(@NonNull Intent intent) {
        super.onNewIntent(intent);

        // check intent
        FlutterAuthUiPlugin.catchEmailLink(this, intent);
    }
}
```

**Comment by koji-1009** [comment]
I released [2.3.1](https://pub.dev/packages/flutter_auth_ui/versions/2.3.1)!
